lib/bundler/cli.rb

```diff
@@ -437,13 +437,22 @@ def env
     # into the corresponding `bundle help #{command}` call
     def self.reformatted_help_args(args)
       bundler_commands = all_commands.keys
+      help_flags = %w(--help -h)
       command = args.select {|a| bundler_commands.include? a }
+      other_flags = args.select {|a| a.include? "--" } - help_flags
+      executables = args - command - help_flags - other_flags
+      new_args = []
       if command.empty?
         abort("Could not find command \"#{args.join(" ")}\".")
+      elsif executables.any?
+        new_args << command.first
+        new_args += executables
+        new_args << "--help"
+        new_args += other_flags
```

Comment:
I assume this is how you're suggesting it would be done:
```
      if command.empty?
        abort("Could not find command \"#{args.join(" ")}\".")
      elsif command.include? "exec"
        args
      else
        ["help", command.first]
      end
```
However only making an exception for `exec` makes `bundle exec --help` throw `bundler: command not found: --help`.